### PR TITLE
Add release notes for 15.1 (Swift bridge announcement)

### PR DESCRIPTION
**Pull request by hot3eed**

### CODE REVIEW

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ABI-stable since version 5, this long-awaited bridge allows Frida to play nicely
+with binaries written in Swift. Whether you [consider][] Swift a static or a
+dynamic language, one thing is for sure, it just got a lot more dynamic with
+this Frida release.
+
+## Metadata
+
+Probably the first thing a reverser does when they start reversing a binary is
+getting to know the different data structures that the binary defines. So it
+made most since to start by building the Swift equivalent of the `ObjC.classes`
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:20]:
```suggestion
made most sense to start by building the Swift equivalent of the `ObjC.classes`
```

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+`TargetTypeContextDescriptor` C++ struct, defined in
+[include/swift/ABI/Metadata.h][] at the time of writing. This data structure
+includes the type name, its fields, its methods (if applicable,) and other useful
+data depending on the type at hand. These data structures are pointed to by
+relative pointers (defined in [include/swift/Basic/RelativePointer.h][].) In
+Mach-O binaries, these are stored in the `__swift5_types` section.
+
+So to dump types, Frida basically iterates over these data structures and
+parses them along the way, very similar to what [dsdump][] does, except that you
+don't have to build the Swift compiler to in order to tinker with it.
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:37]:
```suggestion
don't have to build the Swift compiler in order to tinker with it.
```

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+Frida also has the advantage of being able to probe into
+internal Apple dylibs written in Swift, and that's because we don't need to
+parse the `dyld_shared_cache` thanks to the private `getsectiondata` API, whch
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:41]:
```suggestion
parse the `dyld_shared_cache` thanks to the private `getsectiondata` API, which
```

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+docs, and it's implemented by the back-end, LLVM.
+
+The process that precedes physical lowering is termed "semantic lowering," which
+is the compiler front-end figuring out who "owns" a value and whether
+it's direct or indirect. Some structs, even though they might be smaller than
+4 registers, have to be passed indirectly, because, for example, they are
+generic and thus their exact memory layout is not known at compile time, or
+because they include a weak reference that has to be in-memory at all times.
+
+Frida had to implement both semantic and physical lowering in order to be able
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:71]:
```suggestion
We had to implement both semantic and physical lowering in order to be able
```

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+the type's metadata to figure out whether we should pass a value directly or
+not.
+
+The compiler [docs][] are a great resource to learn more about the calling
+convention.
+
+## Interception
+
+Because Swift passes structs directly in registers, there isn't a 1-to-1 mapping
+between registers and actual arguments, as is the case for SystemV.
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:84]:
```suggestion
between registers and actual arguments.
```

Structs passed by value might end up sliced across registers with SystemV as well – so maybe it's better to just drop that last part?

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+And now that we have JavaScript wrappers for types, and are able to call Swift
+functions from the JS runtime, a good next step would be extending `Interceptor`
+to support Swift functions.
+
+For functions that are not stripped, we use a simple regex to parse argment
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:90]:
```suggestion
For functions that are not stripped, we use a simple regex to parse argument
```

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+types and names, same for return values. After parsing them we retrieve the
+type metadata, figure the type's layout, then simply construct JS wrappers
```
Comment (oleavr) [inline _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown:92]:
```suggestion
type metadata, figure out the type's layout, then simply construct JS wrappers
```

**Review by oleavr — CHANGES_REQUESTED** [review]
Awesome!! 🤘

File: _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown (hunk 5 above)
Comment (hot3eed) [reply _i18n/en/_posts/2021-09-03-frida-15-1-released.markdown]: Ah, interesting. Definitely yeah, let's drop it.

**Pull request merged**